### PR TITLE
iOS diff review: changed-files list + full-screen hunk-by-hunk file diffs

### PR DESCRIPTION
**Pull request by lawrencecchen**
Native mobile diff review, built to Mitchell Hashimoto's spec for what mobile code review should be (his grading of existing tools: codex diff review C+ on desktop, F on mobile): a changed-files list you can scan WITHOUT diffs, full-screen per-file diffs (not a drawer), hunk-by-hunk navigation with swipe, a bookmark to tag a line and jump back after browsing elsewhere, and real finger targets.

**Mac side** — two new workspace-scoped RPC verbs:
- `mobile.workspace.diff_status`: changed files with status + `+n −m` counts. Two separate `git diff HEAD` invocations (`--numstat -z` and `--name-status -z`) merged by path (combining them in one invocation silently suppresses numstat), plus `ls-files --others` for untracked.
- `mobile.workspace.diff_file`: per-file unified diff, untracked via `git diff --no-index`, capped at 2 MiB on UTF-8 character boundaries with a `truncated` flag (mobile frame cap is 8 MiB).
- Bodies are async and run git off the main actor at utility priority (no typing-latency impact); registered in the TerminalController RPC switch, control-socket mobile-host alias, and MobileHostService auth gate. Remote/SSH workspaces return a clear error in v1.

**Shared** — new `Packages/Shared/CmuxDiffModel` package: pure `UnifiedDiffParser` (multi-hunk, renames, no-index, binary, trailing-newline, truncation cases tested), parsed on-device so the wire carries plain unified diff text. Added to the CI packages list.

**iOS** — `DiffReview/` in CmuxMobileShellUI: files list (pull-to-refresh, empty state, ≥44pt rows), full-screen file view rendering hunks with the agent-chat diff card conventions, prev/next controls + swipe crossing file boundaries, hunk counter, haptics on hunk change, bookmark with jump-back chip (`bookmark.fill` state). `DiffReviewSession` model unit-tested (navigation, bookmark, unknown-vs-zero hunk-count guard). Entry: "Review Changes" in the workspace toolbar menu. EN+JA localized.

**Known follow-ups** (deliberate v1 scope): the control-socket debug/CLI alias lane still runs git synchronously on the main actor (phone data plane does not); "Review Changes" is not yet gated on the `workspace.diff.v1` capability against older Macs; comments/annotations on hunks are a future round.

**Tests:** CmuxDiffModel 7/7, CmuxGit 66/66 (numstat/name-status fixtures for renames, binary, untracked), MobileRPC DTO decode, DiffReviewSession suite, control-socket coordinator tests, workspace-groups check all green.

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/7570"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1786067636&installation_id=133855650&pr_number=7570&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F7570&signature=8eca14a01829dba86e8eca55819c9ff3c4ddc4b195795df5c032376c05eb18e5"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

---

> [!NOTE]
> **Medium Risk**
> Touches workspace-scoped source reads, git subprocess execution, and mobile auth routing; mitigated by caps, worker-lane isolation, and fail-closed path validation, but large surface area and edge-case git behavior warrant careful review.
> 
> **Overview**
> Adds **native iOS “Review Changes”** for local workspaces when the paired Mac advertises **`workspace.diff.v1`**: a changed-files list and full-screen, hunk-by-hunk unified diffs with swipe navigation, bookmarks, jump-back, truncation banners, and localized RPC error handling.
> 
> **Mac / transport:** New **`CmuxGit`** `GitDiffService` runs bounded, sandboxed git subprocesses (output caps, deadlines, env scrubbing, single-file and rename validation). **`mobile.workspace.diff_status`** and **`mobile.workspace.diff_file`** are routed on the **control-socket worker lane** (not main actor). Host capabilities and **attach-ticket workspace auth** gate diff reads.
> 
> **Shared + mobile plumbing:** New **`CmuxDiffModel`** with **`UnifiedDiffParser`** (parsed off-main on device). iOS gains typed RPC decoders, **`MobileShellComposite`** fetch helpers, workspace-list **remote/tmux eligibility** flags, and ticket coverage mirroring the host.
> 
> **CI:** Builds/tests **`CmuxDiffModel`** and runs package-resolved policy tests before the existing checker.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e1c0df79a1b6cccdc5a72afbc15b9f1b55ab4dac. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Adds native iOS diff review for local workspaces when the paired Mac advertises `workspace.diff.v1`: a changed‑files list and full‑screen, hunk‑by‑hunk diffs with fast navigation, retry, and clear truncation states. Git runs on the socket‑worker lane with watchdogs and cancellation; parsing runs off‑main, and diffs are bound to repository snapshot tokens with stale‑repo reload+retry.

- **New Features**
  - iOS: changed‑files list and full‑screen unified diffs with swipe (cross‑file), haptics, bookmark/jump‑back, file switcher + search, Retry on failure, a “File list truncated” banner, and a 2,000‑lines‑per‑hunk render cap (EN/JA). Loads are path‑scoped; stale parses are cancelled.
  - Entry: “Review Changes” appears only when the owner Mac supports `workspace.diff.v1` and the workspace is local; SSH and tmux mirrors are ineligible. Menu gating is per‑workspace.
  - Shared: new `CmuxDiffModel` with a pure `UnifiedDiffParser`; parsing runs off the main thread and is covered by CI.

- **Bug Fixes**
  - Transport/runtime: diff RPCs run on the control‑socket worker lane; per‑listing/file output is byte‑bounded (2 MiB lists, UTF‑8 trimming) with `truncated` flags. A cancellable watchdog terminates stalled git; RPC cancellation propagates into detached work.
  - Auth/capabilities: diff RPCs require attach‑ticket workspace scope; the iOS client mirrors ticket coverage and falls back to Stack auth. iOS decoders map remote/tmux eligibility and optional `truncated`.
  - Bounds/integrity: bind file diffs to a `snapshot_token` with stale‑repo reload+retry; enforce single‑file sections and required rename headers; reject directory/glob requests; wrap pathspecs in `:(literal)` and add `--` before `--no-index`; scrub `GIT_DIR`/`GIT_WORK_TREE`/`GIT_INDEX_FILE`/`GIT_EXTERNAL_DIFF`; pass `--no-ext-diff --no-textconv`. Clears stale bookmarks.

<sup>Written for commit 60e364f317644dc3cabae5ffc721ddb8b77a191e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/7570?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added a workspace “diff review” experience with a changed-files list and unified-diff viewer, including hunk navigation, jump-back, and bookmarking.
  * Introduced mobile diff RPC support with typed decoding, plus git-backed diff listing and unified-diff retrieval.
* **Bug Fixes**
  * Improved diff parsing and rendering for empty/truncated diffs, renamed/untracked/binary cases, and trailing-newline edge cases; improved cancellation safety and truncation handling.
  * Tightened diff-request authorization/routing and added diff capability discovery.
* **Tests / CI**
  * Added/expanded unit and UI test coverage for diff parsing, RPC decoding, navigation behavior, and updated CI to run CmuxDiffModel tests.

### CODE REVIEW

**Comment by lawrencecchen** [comment]
@codex review

File: Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        let pipe = Pipe()
+        process.standardOutput = pipe
+        process.standardError = FileHandle.nullDevice
+        do {
+            try process.run()
+            let data = pipe.fileHandleForReading.readDataToEndOfFileOrEmpty()
+            process.waitUntilExit()
+            guard acceptedTerminationStatuses.contains(process.terminationStatus) else {
+                return GitProcessResult(output: nil)
+            }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift:436]: Rejected: plain `git diff` (no `--exit-code`/`--quiet`) exits 0 even when there are differences, so the `changedFiles` and tracked `fileDiff` invocations never see exit 1 on changes; only `git diff --no-index` exits 1 on differences, and that call already passes `acceptedTerminationStatuses: [0, 1]`. Real git failures exit 128/129.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewSession.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    func moveBackward() {
+        guard canMoveBackward else { return }
+        if currentHunkIndex > 0 {
+            currentHunkIndex -= 1
+        } else {
+            currentFileIndex -= 1
+            let previousCount = currentHunkCount
+            currentHunkIndex = max(0, previousCount - 1)
+        }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewSession.swift:163]: Fixed in 32082109aa: backward navigation into a file whose hunk count is not loaded yet records a pending seek, and `recordHunkCount` lands on that file's last hunk when the count arrives; any explicit navigation cancels the pending seek. Unit tests added.

*— Claude Code*

File: Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    ///   - repoRoot: Repository root.
+    ///   - path: Repository-relative path.
+    /// - Returns: Raw one-file unified diff, or `nil` when git fails.
+    public func fileDiff(repoRoot: String, path: String) -> GitFileDiff? {
+        let normalizedPath = path.trimmingCharacters(in: .whitespacesAndNewlines)
+        guard !normalizedPath.isEmpty else { return nil }
+        if isUntracked(repoRoot: repoRoot, path: normalizedPath) {
+            let result = runGit(
+                in: repoRoot,
+                arguments: ["diff", "--no-index", "--no-color", "/dev/null", normalizedPath],
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in 32082109aa: the `--no-index` invocation now passes `--` before `/dev/null` and the path, with a real-git regression test covering an untracked `-flag.swift`.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/WorkspaceDetailView.swift [hunk 4]
```diff
@@ -427,6 +439,14 @@ struct WorkspaceDetailView: View {
             // Only while the terminal pane is showing: browser and chat modes
             // do not mount a terminal surface for text capture.
             if activeBrowser == nil && !isChatMode {
+                Button(action: openDiffReviewFromMenu) {
+                    Label(
+                        L10n.string("mobile.diff.reviewChanges", defaultValue: "Review Changes"),
+                        systemImage: "doc.text.magnifyingglass"
+                    )
+                }
+                .accessibilityIdentifier("MobileReviewChangesMenuItem")
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/WorkspaceDetailView.swift]: Fixed in 32082109aa: the Review Changes menu item is now gated on the owning Mac advertising `workspace.diff.v1` via `MobileShellComposite.supportsDiffReview(for:)`, checked per workspace (foreground or secondary Mac) like the other capability-gated workspace controls.

*— Claude Code*

[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/WorkspaceDetailView.swift]: Fixed in 32082109aa (this review ran against the earlier commit): the Review Changes menu item is gated on the owning Mac advertising `workspace.diff.v1` via `MobileShellComposite.supportsDiffReview(for:)`, checked per workspace like the other capability-gated controls.

*— Claude Code*

File: Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in 32082109aa (this review ran against the earlier commit): the `--no-index` call now passes `--` before `/dev/null` and the path, with a real-git regression test for an untracked `-flag.swift`.

*— Claude Code*

File: Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift [hunk 5]
```diff
@@ -311,6 +312,20 @@ public final class MobileShellComposite: MobileTerminalOutputSinking {
...
     public var supportsDogfoodFeedback: Bool { supportedHostCapabilities.contains(Self.dogfoodFeedbackCapability) }
+    /// Whether the Mac that owns `workspaceID` advertises native diff review
+    /// (`workspace.diff.v1`). Checked per workspace because the diff RPCs route
+    /// to the owning Mac (foreground or secondary), whose capability set can
+    /// differ from the foreground's.
+    public func supportsDiffReview(for workspaceID: MobileWorkspacePreview.ID) -> Bool {
+        let owner = workspaces.first(where: { $0.id == workspaceID })?.macDeviceID
+        if owner == nil || owner == foregroundMacDeviceID || owner == Self.foregroundAnonymousKey {
+            return supportedHostCapabilities.contains(Self.workspaceDiffCapability)
+        }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift]: Rejected: the nil-owner branch intentionally mirrors `workspaceMutationTarget(for:)`, which routes a nil-owner workspace's RPCs to the FOREGROUND client, so the capability consulted is exactly the Mac that will serve the request. `macDeviceID` is tagged phone-side during aggregation (nil means foreground-fetched, not 'old Mac'); failing closed here would hide the action for foreground workspaces while their RPCs would succeed.

*— Claude Code*

File: Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift [hunk 6]
```diff
@@ -320,6 +321,20 @@ public final class MobileShellComposite: MobileTerminalOutputSinking {
...
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift]: Rejected, same rationale as the sibling thread: the gate matches workspaceMutationTarget's routing precedence, so capability and dispatch target cannot diverge.

*— Claude Code*

File: Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift [hunk 7]
```diff
@@ -320,6 +321,20 @@ public final class MobileShellComposite: MobileTerminalOutputSinking {
...
+        if let owner, let subscription = secondaryMacSubscriptions[owner] {
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift]: Rejected, same rationale as the sibling thread: the gate matches workspaceMutationTarget's routing precedence, so capability and dispatch target cannot diverge.

*— Claude Code*

File: Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShell/Sources/CmuxMobileShell/MobileShellComposite.swift]: Rejected, same rationale as the sibling thread: the gate matches workspaceMutationTarget's routing precedence, so capability and dispatch target cannot diverge.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewFilesView.swift [hunk 8]
```diff
@@ -0,0 +1,149 @@
+import CmuxMobileRPC
+import CmuxMobileSupport
+import SwiftUI
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewFilesView.swift]: Fixed in 079c98f305: CmuxMobileRPC is now a direct package and target dependency of CmuxMobileShellUI (test target too), with the lockfile originHash refreshed.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewSession.swift [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+    private var currentHunkCountIfLoaded: Int? {
+        guard let path = currentFile?.path else { return nil }
+        return hunkCountsByPath[path]
+    }
+
+    var hasJumpBackTarget: Bool {
+        guard let bookmark, let currentFile else { return false }
+        return bookmark.filePath != currentFile.path || bookmark.hunkIndex != currentHunkIndex
+    }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewSession.swift:45]: Fixed in 079c98f305: setFiles clears a bookmark whose file is no longer in the list, with a unit test.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewFileView.swift [hunk 10]
```diff
@@ -0,0 +1,305 @@
...
+            isTruncated = result.isTruncated
+            session.recordHunkCount(result.hunks.count, for: path)
+        } catch is CancellationError {
+            return
+        } catch {
+            guard loadedPath == path, !Task.isCancelled else { return }
+            hunks = []
+            isTruncated = false
+            session.recordHunkCount(0, for: path)
+            errorMessage = error.localizedDescription
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewFileView.swift]: Rejected as a deliberate choice: recording a zero count for a failed load keeps file navigation available so the user can move past a file whose diff cannot load (the error state with a Retry button is shown in place). Leaving the count unknown would disable canMoveForward and strand the user on the broken file; a successful Retry replaces the count.

*— Claude Code*

File: Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+            return nil
+        }
+        if isUntracked(repoRoot: repoRoot, path: path, maxOutputBytes: maxOutputBytes) {
+            let result = runGit(
+                in: repoRoot,
+                // `--no-ext-diff --no-textconv`: this output feeds a
+                // machine parser, so repo/user-configured `diff.external`
+                // or textconv drivers must neither execute nor replace the
+                // unified format.
+                arguments: ["diff", "--no-index", "--no-ext-diff", "--no-textconv", "--no-color", "--", "/dev/null", path],
```
[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in c9b3dfec80. Bare dash uses the git-side spelling ./- while the API path remains -. Regression coverage is in e56013d389.

*— Claude Code*

[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in c9b3dfec80. Bare dash uses the git-side spelling ./- while the API path remains -. Regression coverage is in e56013d389.

*— Claude Code*

[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in c9b3dfec80. Bare dash uses the git-side spelling ./- while the API path remains -. Regression coverage is in e56013d389.

*— Claude Code*

[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in c9b3dfec80. Bare dash uses the git-side spelling ./- while the API path remains -. Regression coverage is in e56013d389.

*— Claude Code*

[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Fixed in c9b3dfec80. Bare dash uses the git-side spelling ./- while the API path remains -. Regression coverage is in e56013d389.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewFileRow.swift [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+        return String(
+            format: L10n.string(
+                "mobile.diff.fileAccessibilityFormat",
+                defaultValue: "%1$@, %2$@, %3$@, %4$@"
+            ),
+            statusAccessibilityText,
+            file.path,
+            additions,
+            deletions
+        )
```
[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewFileRow.swift:157]: Fixed in c046703d2c. File-row accessibility now uses explicit singular/other localized keys with interpolation, and the composite label uses localized interpolation. English and Japanese catalog entries were updated.

*— Claude Code*

File: Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewHunkView.swift [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+                "mobile.diff.hunkAccessibilityFormat",
+                defaultValue: "Hunk %1$d of %2$d. Old lines %3$d through %4$d. New lines %5$d through %6$d"
+            ),
+            position,
+            total,
+            hunk.oldStart,
+            max(hunk.oldStart, hunk.oldStart + hunk.oldCount - 1),
+            hunk.newStart,
+            max(hunk.newStart, hunk.newStart + hunk.newCount - 1)
+        )
```
[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/iOS/CmuxMobileShellUI/Sources/CmuxMobileShellUI/DiffReview/DiffReviewHunkView.swift:80]: Fixed in c046703d2c. Zero-count hunk sides now announce localized “No old lines” or “No new lines”; nonzero sides use localized ranges. English and Japanese catalog entries were updated.

*— Claude Code*

File: Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift [hunk 14]
```diff
@@ -0,0 +1,489 @@
...
+            path = pieces[1]
+            index += 1
+        } else {
+            guard index + 1 < tokens.count else { return nil }
+            oldPath = nil
+            path = tokens[index + 1]
+            index += 2
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (azooz2003-bit) [reply Packages/macOS/CmuxGit/Sources/CmuxGit/Diff/GitDiffService.swift]: Addressed in 82629d2e6d while fixing the Swift file-length gate. Bounded subprocess execution now lives in `GitProcessRunner.swift`; `GitDiffService` remains focused on diff/status orchestration and parsing.

*— Claude Code*
